src/libexpr/eval-inline.hh

```diff
@@ -32,39 +32,39 @@ LocalNoInlineNoReturn(void throwTypeError(const Pos & pos, const char * s, const
 
 void EvalState::forceValue(Value & v, const Pos & pos)
 {
-    if (v.type == tThunk) {
+    if (v.isThunk()) {
         Env * env = v.thunk.env;
         Expr * expr = v.thunk.expr;
         try {
-            v.type = tBlackhole;
+            v.setBlackhole();
             //checkInterrupt();
             expr->eval(*this, *env, v);
         } catch (...) {
-            v.type = tThunk;
+            v.setThunk();
             v.thunk.env = env;
             v.thunk.expr = expr;
```

Comment:
If we're refactoring this stuff anyway, then it's better to have `v.setThunk()` take arguments to initialize the value properly, i.e.
```
v.mkThunk(env, expr);
```

(Also `mkThunk` instead of `setThunk` for consistency with all the existing `mk...` functions.)

Reply:
Hiding the initialisation of thunks and other types is a good thing anyway if we ever want to revive the parallel evaluator, since (IIRC) it required fields to be set in a particular order. E.g. when transitioning from tBlackhole -> tString, you have to set the `s` field *before* setting type to `tString`, since otherwise another thread might think the value is initialized.

Reply:
Sounds good, I'll create `v.mk<Type>` methods. I can also remove the global `mkInt(Value & v, NixInt n)` functions (though e.g. `EvalState::mkAttrs` should stay since it initializes memory)

Reply:
I left most static `mk*` functions around since I don't have motivation to refactor this as well. But this could easily be done later.